### PR TITLE
Fix #4: predict-eth2.md: do simple model-based prediction

### LINKED ISSUE #4: predict-eth2.md: do simple model-based prediction
- [ ] In eth-predict2.md, do a simple model-based prediction (eg linear model from scikit learn). As part of this: write the code that uses just the 1 provided dataset. Put this into eth-predict2.md

Parent epic: [predict-eth#3](https://github.com/oceanprotocol/predict-eth/issues/3)

### PR DESCRIPTION
**Pull request by jfdelgad**
Fixes # 4 .

Changes proposed in this PR:

- Fixes #4: Changes the demo ML model for a basic linear model

### CODE REVIEW

**Comment by trentmc** [comment]
Jaime, in your PRs, please
- make the title have the # and name of issue you're fixing
- make the description have the # of the issue you're fixing

E.g. for this issue:
- title: "Fix #4: predict-eth2.md: do simple model-based prediction"
- description: "Fix #4: <rest of descr>"

File: predict-eth2.md [hunk 1]
```diff
@@ -89,20 +89,29 @@ print_datetime_info("CEX data info", allcex_uts)
...
 ### 3.2  Run the AI model to make future ETH price predictions
 
 Predictions must be one prediction every hour on the hour, for a 12h period: Oct 3 at 1am, 2am, 3am, 4am, 5am, 6am, 7am, 8am, 9am, 10am, 11am, 12pm (UTC). Therefore there are 12 predictions total.
 
 In the same Python console:
 ```python
+start_dt = datetime.datetime(2022, 10, 3, 1, 00)
+target_uts = ph.target_12h_unixtimes(start_dt)
+tmp = [datetime.datetime.utcfromtimestamp(x) for x in target_uts]
+future_inputs = pd.DataFrame({"ds": tmp})
```
Comment (trentmc) [inline predict-eth2.md:109]: Missing an import for pandas

File: predict-eth2.md [hunk 2]
```diff
@@ -89,20 +89,29 @@ print_datetime_info("CEX data info", allcex_uts)
...
 
 ### 3.2  Run the AI model to make future ETH price predictions
 
 Predictions must be one prediction every hour on the hour, for a 12h period: Oct 3 at 1am, 2am, 3am, 4am, 5am, 6am, 7am, 8am, 9am, 10am, 11am, 12pm (UTC). Therefore there are 12 predictions total.
 
 In the same Python console:
 ```python
+start_dt = datetime.datetime(2022, 10, 3, 1, 00)
+target_uts = ph.target_12h_unixtimes(start_dt)
+tmp = [datetime.datetime.utcfromtimestamp(x) for x in target_uts]
```
Comment (trentmc) [inline predict-eth2.md:108]: Replace with:

``python
target_dts = to_datetimes(target_uts)
``
In general, please use the support methods at the bottom. That's what they are there for:) And they help readability.

**Review by trentmc — CHANGES_REQUESTED** [review]
A few small changes requested

File: predict-eth2.md (hunk 1 above)
Comment (jfdelgad) [reply predict-eth2.md]: Fixed

File: predict-eth2.md (hunk 2 above)
Comment (jfdelgad) [reply predict-eth2.md]: Fixed

File: predict-eth2.md [hunk 3]
```diff
@@ -89,20 +89,31 @@ print_datetime_info("CEX data info", allcex_uts)
 ### 3.1  Build a simple AI model
 
 Here's where you build whatever AI/ML model you want, leveraging the data from the previous step.
+```python
+import pandas as pd
+from prophet import Prophet
+uts = [datetime.datetime.utcfromtimestamp(x) for x in allcex_uts]
```
Comment (trentmc) [inline predict-eth2.md:95]: > uts = [datetime.datetime.utcfromtimestamp(x) for x in allcex_uts]

Confusing. This looks like "uts" to "uts" conversion? Yet the new variable took out the "allcex_" part.

Please rename to be less confusing. And create/use support method in Appendix, in a pattern similar to others, eg. `to_unixtimes()`.

File: predict-eth2.md [hunk 4]
```diff
@@ -89,20 +89,31 @@ print_datetime_info("CEX data info", allcex_uts)
 ### 3.1  Build a simple AI model
 
 Here's where you build whatever AI/ML model you want, leveraging the data from the previous step.
+```python
+import pandas as pd
+from prophet import Prophet
```
Comment (trentmc) [inline predict-eth2.md:94]: To be consistent with rest of this README, please put the imports in the Appendix.

File: predict-eth2.md [hunk 5]
```diff
@@ -89,20 +89,31 @@ print_datetime_info("CEX data info", allcex_uts)
...
+future_inputs = pd.DataFrame({"ds": target_dts})
+
 #get predicted ETH values
-import random
-avg = 1300
-rng = 25.0
-pred_vals = [avg + rng * (random.random() - 0.5) for i in range(12)]
+forecast = model.predict(future_inputs)
+result = forecast.set_index('ds')['yhat'][-12:]
+pred_vals = result.to_numpy()
```
Comment (trentmc) [inline predict-eth2.md:116]: I recommend to change to a single line: 
```python
pred_vals = forecast.set_index('ds')['yhat'][-12:].to_numpy()
```

**Review by trentmc — CHANGES_REQUESTED** [review]

File: predict-eth2.md (hunk 4 above)
Comment (jfdelgad) [reply predict-eth2.md:94]: Done

File: predict-eth2.md (hunk 3 above)
Comment (jfdelgad) [reply predict-eth2.md:95]: changed to use `to_datetimes`, new variable name `dts`

File: predict-eth2.md (hunk 5 above)
Comment (jfdelgad) [reply predict-eth2.md:116]: Done

**Review by trentmc — APPROVED** [review]

**Comment by trentmc** [comment]
Hi @jfdelgad , given the changes to the structure of the READMEs, this PR will need to get changed heavily, such that it's one of the examples.

**Comment by jfdelgad** [comment]
> Hi @jfdelgad , given the changes to the structure of the READMEs, this PR will need to get changed heavily, such that it's one of the examples.

This has been addressed, i added you as reviewer of the example. I am closing this pull request.

**Pull request closed without merging**